### ISSUE TITLE
Quarto projects don't share dependencies

### ISSUE THREAD

**Issue by hadley** [quarto-dev/quarto-cli#9974]
### Bug description

When rendering a project, I would expect HTML output to share dependencies, rather than each dependency getting a complete copy.

### Steps to reproduce

1. Create a new directory
2. Make it a quarto project by adding `_quarto.yaml` and add two qmd files: `touch _quarto.yaml file1.qmd file2.qmd`
3. Run `quarto render`
4. Observe that both `file1_files` and `file_files2` contain a copy of the same dependencies

### Expected behavior

There should only be one copy of the dependencies.

### Actual behavior

There are two copies of the dependencies

### Quarto check output

```bash
Quarto 1.5.40
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.2.0: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
      Typst version 0.11.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.5.40
      Path: /Applications/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: (external install)
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: TinyTex
      Path: /Users/hadleywickham/Library/TinyTeX/bin/universal-darwin
      Version: 2024

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.12.3
      Path: /opt/homebrew/opt/python@3.12/bin/python3.12
      Jupyter: (None)

      Jupyter is not available in this Python installation.
      Install with python3 -m pip install jupyter

[✓] Checking R installation...........OK
      Version: 4.4.0
      Path: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources
      LibPaths:
        - /Users/hadleywickham/Library/R/arm64/4.4/library
        - /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/library
      knitr: 1.47
      rmarkdown: 2.27

[✓] Checking Knitr engine render......OK
```

**Comment by cscheid** [comment]
@dragonstyle @jjallaire This only happens in `type: default` Quarto projects (as opposed to `book` or `website`). Was that by design or something that we could fix?

**Comment by aisi-inspect** [comment]
Yes, this is by design. `default` is necessarily minimal in its assumptions. You might e.g. be creating 20 documents split across 5 (or 20) directories, each of which is self-contained (i.e. has no parent directory dependencies). 

So at a baseline minimum we need a mode that doesn't automatically do this. We could improve this in 1 of 3 ways:

1) Introduce an option to `default` which turns this on;
2) Add another project type which is intended as a base for custom project types which want to share dependencies; or
3) Decouple `website` from bootstrap so you'd say `website` (and that would bundle dips) then we'd feed you a bunch of nav data structures, etc.)

(3) is by far the most work (like, a major feature for a major release). We deemed this coupling acceptable b/c you certainly produce any imaginable website you want with bootstrap and we aren't trying to compete w/ Hugo and Jeckel, more trying to be a flexible but opinionated way to make scientific websites. 

I'd probably argue for (1). It's some non trivial amount of work to make sure that everywhere we check for dep target paths we also include this case but pretty manageable.

**Comment by jjallaire** [comment]
Yes, this is by design. `default` is necessarily minimal in its assumptions. You might e.g. be creating 20 documents split across 5 (or 20) directories, each of which is self-contained (i.e. has no parent directory dependencies). 

So at a baseline minimum we need a mode that doesn't automatically do this. We could improve this in 1 of 3 ways:

1) Introduce an option to `default` which turns this on;
2) Add another project type which is intended as a base for custom project types which want to share dependencies; or
3) Decouple `website` from bootstrap so you'd say `website` (and that would bundle dips) then we'd feed you a bunch of nav data structures, etc.)

(3) is by far the most work (like, a major feature for a major release). We deemed this coupling acceptable b/c you certainly produce any imaginable website you want with bootstrap and we aren't trying to compete w/ Hugo and Jeckel, more trying to be a flexible but opinionated way to make scientific websites. 

I'd probably argue for (1). It's some non trivial amount of work to make sure that everywhere we check for dep target paths we also include this case but pretty manageable.
